engine/consensus/compliance/core.go

```diff
@@ -227,25 +232,48 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Bloc
 // processed as well.
 // No errors are expected during normal operation. All returned exceptions
 // are potential symptoms of internal state corruption and should be fatal.
-func (c *Core) processBlockAndDescendants(proposal *messages.BlockProposal) error {
+func (c *Core) processBlockAndDescendants(proposal *messages.BlockProposal, parent *flow.Header) error {
 	blockID := proposal.Header.ID()
 
+	log := c.log.With().
+		Str("block_id", blockID.String()).
+		Uint64("block_height", proposal.Header.Height).
+		Uint64("block_view", proposal.Header.View).
+		Uint64("parent_view", parent.View).
+		Logger()
+
 	// process block itself
-	err := c.processBlockProposal(proposal)
+	err := c.processBlockProposal(proposal, parent)
 	if err != nil {
 		if engine.IsOutdatedInputError(err) {
 			// child is outdated by the time we started processing it
 			// => node was probably behind and is catching up. Log as warning
-			c.log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
+			log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
 			return nil
-		}
-		if engine.IsInvalidInputError(err) {
-			// the block is invalid; log as error as we desire honest participation
-			// ToDo: potential slashing
-			c.log.Warn().Err(err).Msg("received invalid block from other node (potential slashing evidence?)")
+		} else if engine.IsInvalidInputError(err) || engine.IsUnverifiableInputError(err) {
```

Comment:
similarly to [my comment before](https://github.com/onflow/flow-go/pull/3294#discussion_r994034443), it might be nice to not duplicate checks here and encapsulate some of the logic in auxiliary functions.

Reply:
https://github.com/onflow/flow-go/pull/3294/commits/a2c8220c13e22450ac7bdbd1df224366bc917212